Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2460 -- Enable showing mix of items & subcategories in sidebar categories

Resolves [EREGCSC-2460](https://jiraent.cms.gov/browse/EREGCSC-2460)

**Description**

(From JIRA story:)
Currently in our regulation page sidebars, public & internal document categories can only display items or subcategories. If there are items that belong to the category and subcategories, only the subcategories are displayed. 

Let's make our sidebar categories (public categories and internal categories) able to show both items and subcategories.

Subcategories first (containing items that belong to the subcategories), then a separator line, then miscellaneous items (that belong to the parent category).

**This pull request changes:**

- Removes `if` directive that only allows items to be displayed under a Category if there is not a Subcategory
- Adds fixture files to be used in end to end tests
- Adds end to end tests to assert that both items and subcategories can coexist under a category

**Steps to manually verify this change:**

1. Green check marks
2. Visit the [experimental deployment ](https://1myqqgkjz7.execute-api.us-east-1.amazonaws.com/dev1298) and poke around Parts, Subparts, etc to make sure everything looks good
3. Find a Part that contains public and internal resources that coexist under a category with subcategories
   - Ex: [42 CFR 433 Subpart A](https://1myqqgkjz7.execute-api.us-east-1.amazonaws.com/dev1298/42/433/Subpart-A/2024-04-23/)
   - Public docs: under Subregulatory Guidance, an item should be visible after the list of subcategories
   - Internal docs: under Internal Documents and Correspondence, to items should be visible after a subcategory
